### PR TITLE
[ci] Move CI workers from `macos-12` to `macos-latest`

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** https://github.com/actions/runner-images/issues/10721

**Changes:**

**Notes for Reviewer:**

### CODE REVIEW

File: .github/workflows/python-ci-full.yml [hunk 1]
```diff
@@ -21,16 +21,16 @@ jobs:
       fail-fast: false
       matrix:
         # TODO: decide on Windows CI coverage
-        os: [ubuntu-22.04, macos-12]
-        # os: [ubuntu-22.04, macos-12, windows-2019]
+        os: [ubuntu-22.04, macos-latest]
+        # os: [ubuntu-22.04, macos-latest, windows-2019]
```
Comment (johnkerl) [inline .github/workflows/python-ci-full.yml:25]: re next line: https://github.com/single-cell-data/TileDB-SOMA/pull/3184

File: .github/workflows/python-packaging.yml [hunk 2]
```diff
@@ -57,7 +57,7 @@ jobs:
             os: ubuntu-20.04
             wheel-name: manylinux2014
           - cibw_build: macosx_x86_64
-            os: macos-12
+            os: macos-latest
```
Comment (bkmartinjr) [inline .github/workflows/python-packaging.yml:60]: is there a reason we don't also move to `ubuntu-latest`?

**Review by bkmartinjr — APPROVED** [review]

File: .github/workflows/python-packaging.yml (hunk 2 above)
Comment (johnkerl) [reply .github/workflows/python-packaging.yml:60]: Great idea @bkmartinjr -- happy to try this as a follow-on PR! :)

Comment (johnkerl) [reply .github/workflows/python-packaging.yml:60]: #3186

**Pull request merged**